render/detailed/summary.go

```diff
@@ -53,33 +64,58 @@ func MakeColumn(id string) Column {
 type NodeSummary struct {
 	ID           string        `json:"id"`
 	Label        string        `json:"label"`
-	Linkable     bool          `json:"linkable"` // Whether this node can be linked-to
+	LabelMinor   string        `json:"label_minor"`
+	Rank         string        `json:"rank"`
+	Shape        string        `json:"shape,omitempty"`
+	Stack        bool          `json:"stack,omitempty"`
+	Linkable     bool          `json:"linkable,omitempty"` // Whether this node can be linked-to
+	Pseudo       bool          `json:"pseudo,omitempty"`
 	Metadata     []MetadataRow `json:"metadata,omitempty"`
 	DockerLabels []MetadataRow `json:"docker_labels,omitempty"`
 	Metrics      []MetricRow   `json:"metrics,omitempty"`
+	Adjacency    report.IDList `json:"adjacency,omitempty"`
 }
 
 // MakeNodeSummary summarizes a node, if possible.
-func MakeNodeSummary(n render.RenderableNode) (NodeSummary, bool) {
-	renderers := map[string]func(report.Node) NodeSummary{
+func MakeNodeSummary(n report.Node) (NodeSummary, bool) {
+	renderers := map[string]func(NodeSummary, report.Node) (NodeSummary, bool){
+		render.Pseudo:         pseudoNodeSummary,
 		report.Process:        processNodeSummary,
 		report.Container:      containerNodeSummary,
 		report.ContainerImage: containerImageNodeSummary,
 		report.Pod:            podNodeSummary,
+		report.Service:        serviceNodeSummary,
 		report.Host:           hostNodeSummary,
```

Comment:
I wonder whether we should have another 'pseudo-topology' (called "Grouped") where we do some simple heuristics, like pulling a label from metadata.  Then use this for grouped topologies that don't have a 'real' node - like process-by-name and container-by-hostname.